### PR TITLE
C679 opensearch

### PR DESCRIPTION
**Pull request by ssickles**
A new module to stand up opensearch domains in AWS.

### CODE REVIEW

File: domain.tf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  cluster_config {
+    dedicated_master_enabled = var.dedicated_master_count != 0
+    dedicated_master_count = var.dedicated_master_count
+    dedicated_master_type = var.dedicated_master_type
+
+    instance_count = var.instance_count
+    instance_type = var.instance_type
+
+    zone_awareness_enabled = local.zone_count > 0
```
Comment (BSick7) [inline domain.tf]:
```suggestion
    zone_awareness_enabled = local.zone_count > 1
```

File: domain.tf [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    zone_awareness_config {
```
Comment (BSick7) [inline domain.tf:35]: You may need to make this dynamic. 
I'm not sure what the provider will do if enabled = false and this stanza is specified.

File: outputs.tf [hunk 3]
```diff
@@ -0,0 +1,14 @@
+output "domain_arn" {
+  value       = aws_elasticsearch_domain.this.arn
+  description = "string ||| ARN of the OpenSearch domain"
+}
+
+output "domain_endpoint" {
```
Comment (BSick7) [inline outputs.tf]: There is an implicit convention I've been following with datastores for outputs that makes it easier to hook up capabilities.

Notably, I've started ensuring that the outputs include `db_protocol`, `db_endpoint`, `db_security_group_id`.
- `db_protocol` would be `https`.

**Review by BSick7 — COMMENTED** [review]
Looks good, but a couple of things to change.
Can you run `terraform fmt` also?

File: domain.tf (hunk 1 above)
Comment (ssickles) [reply domain.tf]: Fixed

File: domain.tf (hunk 2 above)
Comment (ssickles) [reply domain.tf:35]: It doesn't seem to have an issue setting zone_awareness_enabled = false while also specifying a zone_awareness_config. When I ran my tests it just ignored the config.

**Pull request merged**